project/MimaExcludes.scala

```diff
@@ -36,9 +36,11 @@ object MimaExcludes {
 
   // Exclude rules for 3.0.x
   lazy val v30excludes = v24excludes ++ Seq(
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.io.SnappyCompressionCodec.version"),
+    // [SPARK-25362][JavaAPI] Replace Spark Optional class with Java Optional
+    ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.api.java.Optional")
     ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.api.java.JavaPairRDD.flatMapValues"),
     ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.streaming.api.java.JavaPairDStream.flatMapValues")
+    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.io.SnappyCompressionCodec.version"),
```

Comment:
And above the line here, add
`// [SPARK-24109] Remove class SnappyOutputStreamWrapper`